testsuite/MDAnalysisTests/parallelism/test_multiprocessing.py

```diff
@@ -0,0 +1,209 @@
+        f = request.param[0]
+        return mda.Universe(f)
+    else:
+        top, trj = request.param
+        return mda.Universe(top, trj)
+
+
+# Define target functions here
+# inside test functions doesn't work
+def cog(u, ag, frame_id):
+    u.trajectory[frame_id]
+
+    return ag.center_of_geometry()
+
+
+def getnames(u, ix):
+    # Check topology stuff works
+    return u.atoms[ix].name
+
+
+def test_trajectory_next(u):
+    u.trajectory.next()
+    u_p = pickle.loads(pickle.dumps(u))
+    u.trajectory.next()
+    u_p.trajectory.next()
+    assert_equal(u.atoms.positions[0], u_p.atoms.positions[0])
+
+
+def test_multiprocess_COG(u):
+    ag = u.atoms[2:5]
+
+    ref = np.array([cog(u, ag, i)
+                    for i in range(3)])
+
+    p = multiprocessing.Pool(2)
+    res = np.array([p.apply(cog, args=(u, ag, i))
+                    for i in range(3)])
+    p.close()
+    assert_equal(ref, res)
+
+
+def test_multiprocess_names(u):
+    if u.trajectory.__class__ in (mda.coordinates.TRJ.NCDFReader,
+                                  mda.coordinates.memory.MemoryReader,
+                                  mda.coordinates.chain.ChainReader):
+        # These Readers contain no information on atom name
+        return True
+    ref = [getnames(u, i)
+           for i in range(3)]
+
+    p = multiprocessing.Pool(2)
+    res = [p.apply(getnames, args=(u, i))
+           for i in range(3)]
+    p.close()
+
+    assert_equal(ref, res)
+
+
+@pytest.fixture(params=[
+    # formatname, filename
+    ('CRD', CRD, dict()),
+    ('DATA', LAMMPSdata_mini, dict(n_atoms=1)),
+    ('DCD', DCD, dict()),
+    ('DMS', DMS, dict()),
+    ('CONFIG', DLP_CONFIG, dict()),
+    ('HISTORY', DLP_HISTORY, dict()),
+    ('INPCRD', INPCRD, dict()),
+    ('LAMMPSDUMP', LAMMPSDUMP, dict()),
+    ('GMS', GMS_ASYMOPT, dict()),
+    ('GRO', GRO, dict()),
+    ('GSD', GSD, dict()),
+    ('MMTF', MMTF, dict()),
+    ('MOL2', mol2_molecules, dict()),
+    ('PDB', PDB_small, dict()),
+    ('PQR', PQR, dict()),
+    ('PDBQT', PDBQT_input, dict()),
+    ('TRR', TRR, dict()),
+    ('TRZ', TRZ, dict(n_atoms=8184)),
+    ('TRJ', TRJ, dict(n_atoms=252)),
+    ('XTC', XTC, dict()),
+    ('XPDB', XPDB_small, dict()),
+    ('XYZ', XYZ_mini, dict()),
+    ('NCDF', NCDF, dict()),
+    ('TXYZ', TXYZ, dict()),
+    ('memory', np.arange(60).reshape(2, 10, 3).astype(np.float64), dict()),
+    ('CHAIN', [GRO, GRO, GRO], dict()),
+    ('CHAIN', [PDB, PDB, PDB], dict()),
+    ('CHAIN', [XTC, XTC, XTC], dict()),
+])
+def ref_reader(request):
+    fmt_name, filename, extras = request.param
+    r = get_reader_for(filename, format=fmt_name)(filename, **extras)
+    try:
+        yield r
+    finally:
+        # make sure file handle is closed afterwards
+        r.close()
+
+
+def test_readers_pickle(ref_reader):
+    ps = pickle.dumps(ref_reader)
+    reanimated = pickle.loads(ps)
+    assert len(ref_reader) == len(reanimated)
+
+
+def test_timestep_pickle(ref_reader):
```

Comment:
do we need to test this for all readers? A single timestep would be enough. There is only a weakref to the readers anyway.

Reply:
Merge it with the test above.